static/js/lib/ckeditor/plugins/MarkdownMediaEmbed.ts

```diff
@@ -0,0 +1,62 @@
+import MediaEmbed from "@ckeditor/ckeditor5-media-embed/src/mediaembed"
+import Plugin from "@ckeditor/ckeditor5-core/src/plugin"
+import Turndown from "turndown"
+import Showdown from "showdown"
+
+import MarkdownSyntaxPlugin from "./MarkdownSyntaxPlugin"
+import { TurndownRule } from "../../../types/ckeditor_markdown"
+
+export const YOUTUBE_SHORTCODE_REGEX = /{{< youtube (\S+) >}}/
+
+export const EMBED_CLASS = "media"
+
+export const YOUTUBE_URL_REGEX = /(?:youtube\.com\/watch\?\S*v=|youtu.be\/)([a-zA-Z0-9_]+)/
+
+class MediaEmbedMarkdownSyntax extends MarkdownSyntaxPlugin {
+  get showdownExtension() {
+    return function mediaEmbedExtension(): Showdown.ShowdownExtension[] {
+      return [
+        {
+          type:    "lang",
+          regex:   YOUTUBE_SHORTCODE_REGEX,
+          replace: (s: string, match: string) =>
+            `<figure class="${EMBED_CLASS}">
+              <oembed url="https://www.youtube.com/watch?v=${match}">
+              </oembed>
+            </figure>`
+        }
+      ]
+    }
+  }
+
+  get turndownRule(): TurndownRule {
+    return {
+      name: "mediaEmbed",
+      rule: {
+        filter:      "figure",
+        replacement: (
+          content: string,
+          node: Turndown.Node,
+          _: Turndown.Options
+        ): string => {
+          const videoId = node
+            ?.querySelector("oembed")
+            ?.getAttribute("url")
+            ?.match(YOUTUBE_URL_REGEX)
+
+          if (videoId && videoId[1]) {
+            return `{{< youtube ${videoId[1]} >}}`
```

Comment:
This is a bit of a nitpick, but I noticed if you put two videos side by side without a line break in between, it renders the shortcodes on the same line:

```
{{< youtube fB9GlZUYNUQ >}}{{< youtube O_XhRdWdphU >}}
```

This seems to render fine in Hugo, I tested it by injecting this into a running course:

![image](https://user-images.githubusercontent.com/12089658/110384524-c14e1100-802b-11eb-9379-f5e81ba3971e.png)

Alternatively, if you insert a line break after the first video, three line breaks seem to be inserted between the shortcodes:

```
{{< youtube fB9GlZUYNUQ >}}

 

{{< youtube O_XhRdWdphU >}}
```

This markdown renders exactly the same in Hugo, so there's no big reason to fix this aside from potential readability.  Ideally there should only be a single line break between the two videos.

Reply:
I added a `\n` at the end of the turndown rule, ensure that youtube videos right next to each other in the editor are on different lines:

![Screenshot from 2021-03-09 14-53-08](https://user-images.githubusercontent.com/6207644/110529441-46e2c700-80e7-11eb-9c1f-190e7ed100e1.png)